plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -554,4 +582,46 @@ public int hashCode()
             throw new UncheckedExecutionException(e);
         }
     }
+
+    private static final class TableStatisticsCacheKey
+    {
+        private final JdbcIdentity identity;
+        private final Map<String, Object> sessionProperties;
```

Comment:
which session properties can affect caching? please add a comment

Reply:
Session property can change the column type, so that was my reason behind this. But in regards to statistics we return double always as estimate, so there is need for this. Thanks